cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkTransactional.java

```diff
@@ -313,6 +324,22 @@ public void onTransactionCompleted(boolean jobSucceeded, @Nullable TransactionFa
     }
 
     @Override
+    public <T extends Dataset> T getDataset(String namespace, String name, Map<String, String> arguments,
+                                            AccessType accessType) throws DatasetInstantiationException {
+      T dataset = datasetCache.getDataset(namespace, name, arguments, accessType);
```

Comment:
I am not a fan of this duplication of code... perhaps it makes more sense to make all getDataset() methods in DynamicDatasetCache require a namespace, and that can be @Nullable.